options.context can provide a function

todda00 opened the pull request:
allow options.context to be a function that accepts context with user, and returns final context.

Since `addCurrentUserToContext(context, loginToken)` is an exported function, I didn't want to break any backwards compatibility with anything using function outside the package, so while this isn't used internally anymore, I kept it as a wrapper to `getUserForContext(loginToken)` and still returns the same results.

todda00 commented:
Reference Issue: #99

lorensr commented:
Looking good, mind adding a test case? Looks like `getUserForContext` doesn't need a `context` param

todda00 commented:
I didn't have context in there at all originally, I think I added it
temporarily when I was moving things around and forgot to remove it again.
I'll clean that up and take a crack at a test case.

On Thu, May 25, 2017 at 2:25 PM, Loren Sands-Ramshaw <
notifications@github.com> wrote:

> Looking good, mind adding a test case? Looks like getUserForContext
> doesn't need a context param
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/apollographql/meteor-integration/pull/100#issuecomment-304085787>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AH-flAR5JabkS7mmrWCaK2nX8DMX68_hks5r9ceagaJpZM4NlkC->
> .
>

todda00 commented:
Cleaned this up and added a test. I had to run the GraphQL endpoint on a unique path for the test, since it passed when running by itself, but failed when running on the same /graphql endpoint with other createApolloServer calls. The options.context was being overwritten by other tests, regardless of it being a function or an object. This seems to be an issue of the way the testing was setup only, since apps wouldn't be (or shouldn't be) running createApolloServer on the same endpoint multiple times.
